[docs] Adds Workflows documentation (#32840)
expo/expo: docs/pages/workflows/control-flow.mdx

---
title: Control flow
description: Learn how to control the flow of Workflows with conditions that run certain jobs based on the result of previous jobs.
---

You can control the flow of your workflows with conditions that run certain jobs based on the result of previous jobs.

## `needs`

> You can add an list of job names to the `needs` keyword on a job. If a job in the list fails, the job will be skipped.

Reviewer: ```suggestion
You can add a list of previous job names using the `needs` keyword on a job. If any job from that list fails, the current job will be skipped.
```

At first it was a bit confusing to read this sentence to quickly understand what was happening. In the last sentence, I am not sure if the current job is skipped or the job failed from the list is skipped.

Assuming that the only previous job names can be added to the job using the `needs` keyword, let's explicitly mention that to improve clarity.

Please correct me if my assumption is wrong and feel free to update the sentence accordingly :)

```yaml .eas/workflows/prerequisite.yaml
jobs:
  build:
    type: build
    params:
      platform: ios
      profile: production
  end-to-end-test:
    # @info #
    needs: [build]
    # @end #
    type: maestro
    params:
      build_id: ${{ jobs.build.outputs.build_id }}
      flow_path: ['./e2e/flow.yaml']
```

## `after`

> You can add a list of job names to the `after` keyword on a job. The jobs in the list will run after the job completes no matter if it succeeded or failed.

Reviewer: ```suggestion
You can add a list of previous job names using the `after` keyword on a job. Any job in the list will run after the current job completes, no matter if it succeeded or failed.
```
